server/controllers/ArticleController.js

```diff
@@ -6,21 +6,166 @@ import models from '../models';
 import numberOfArticles from '../helpers/numberOfArticles';
 
 const {
-  Article, User, ArticleLike, Tag, ArticleTag,
+  Article, User, ArticleLike, Tag, ArticleTag, Bookmark
 } = models;
 const { Op } = Sequelize;
 /**
  * @class ArticleController
  * @description Article related Operations
  */
 class ArticleController {
-/**
+  /**
+ * @description
+ * @param {object} req - request object
+ * @param {object} res - response object
+ * @returns {object} - bookmarks an Article
+ */
+  static bookmarkArticle(req, res) {
+    const userId = req.userData.id;
+    const { articleId } = req.params;
+    Bookmark.findOrCreate({
+      where: {
+        userId,
+        articleId
+      }
+    })
+      .spread((returnedBookmarkedArticle, status) => {
+        if (!status) {
+          return res.status(403).json({
+            status: 'failure',
+            errors: {
+              message: ['you have already bookmarked this article']
+            }
+          });
+        }
+        if (returnedBookmarkedArticle) {
+          req.params.slug = articleId;
+          ArticleController
+            .getAnArticle(req, res, 'Article successfully bookmarked');
+        }
+      });
+  }
+
+  /**
+ * @description
+ * @param {object} req - request object
+ * @param {object} res - response object
+ * @returns {object} - bookmarks an Article
+ */
+  static unbookmarkArticle(req, res) {
+    const userId = req.userData.id;
+    const { articleId } = req.params;
+    Bookmark.findOne({
+      where: {
+        userId,
+        articleId
+      },
+      include: [{
```

Comment:
I observed getArticleBookmarked is not sorted. The bookmark should be sorted in the order of createdAt field. Such that when a user runs a query to get all articles bookmarked, he or she would get the list sorted by createdAt. Thinking about user experience, this is a necessity.